pkgs/development/python-modules/pyarrow/default.nix

```diff
@@ -35,7 +36,8 @@ buildPythonPackage rec {
   sourceRoot = "apache-arrow-${version}/python";
 
   nativeBuildInputs = [ cmake cython pkg-config setuptools-scm ];
-  propagatedBuildInputs = [ numpy six cloudpickle scipy fsspec cffi ];
+  propagatedBuildInputs = [ numpy six cloudpickle scipy fsspec cffi ]
+    ++ lib.optionals _arrow-cpp.cudaSupport [ numba ];
```

Comment:
I don't like that. If we have pyarrow with and without cuda support in a python env things will break in strange and mysterious ways.

Reply:
I'm not sure I follow what the concern is. Are you questioning adding CUDA support at all, or is there a different approach you'd like me to take here?

Reply:
I don't like that the package has multiple build variants which could be mixed together. Can we rely on config.cudaSupport instead of manually overriding the input?

Reply:
IIUC, the `PYARROW_WITH_CUDA=1` bit actually affects the build output, so I'd agree with Sandro:

You can add a `cudaSupport ? config.cudaSupport or false` argument to `pyarrow` and add a consistency check: `assert cudaSupport -> _arrow-cpp.cudaSupport;`